app/controllers/merchant/items_controller.rb

```diff
@@ -1,12 +1,85 @@
 class Merchant::ItemsController < ApplicationController
   before_action :require_merchant
 
+  def new
+    @item = Item.new
+  end
+  
+  def create
+    @item = find_merchant_from_user.items.create(item_params)
+    item_check_creation(@item)
+  end
+
   def index
-    @merchant = current_user.merchants[0]
+    @merchant = find_merchant_from_user
+  end
+
+  def edit
+    @item = Item.find(params[:id])
+  end
+
+  def update
+    merchant = find_merchant_from_user
+    if params[:type]
+      change_item_status(merchant)
+    else 
+      update_item
+    end
+  end
+
+  def destroy
+    merchant = find_merchant_from_user
+    flash[:notice] = "#{find_item(merchant).name} was successfully deleted"
+    find_item(merchant).delete
+    redirect_to '/merchant/items'
   end
 
   private
 
+  def item_params
+    params.permit(:name, :description, :price, :image, :inventory)
+  end
+  
+  def find_item(merchant)
+    merchant.items.find(params[:id])
+  end
+
+  def update_item
+    @item = Item.find(params[:id])
+    @item.update(item_params)
+    if @item.save
+      redirect_to '/merchant/items'
+    else
+      flash[:notice] = @item.errors.full_messages.to_sentence
+      redirect_to "/merchant/items/#{@item.id}/edit"
+    end
+  end
+
+  def change_item_status(merchant)
```

Comment:
this one is to change the item's status from deactivate to activate, used query params here